### PR TITLE
fix: set contents of complementary sidebar based on attention

### LINKED ISSUE #7650: [deck] comp. sidebar should close when soloing an unrelated plank.

### LINKED ISSUE #7825: [composer] complementary sidebar content always based on attention
complementary sidebar contents can currently be out of sync with the main deck area, it should always set its content based on attention to ensure it stays in sync

### PR DESCRIPTION
**Pull request by wittjosiah**
Resolves #7650
Resolves #7825

### CODE REVIEW

File: packages/plugins/plugin-deck/src/components/DeckLayout/ComplementarySidebar.tsx [hunk 1]
```diff
@@ -18,41 +19,45 @@ import { useNode, useNodeActionExpander } from '../../hooks';
...
-        <div role='none' className={mx(deckGrid, 'grid-cols-1 bs-full')}>
-          <NodePlankHeading
-            node={node}
-            id={id}
-            layoutParts={layoutParts}
-            layoutPart='complementary'
-            popoverAnchorId={popoverAnchorId}
-            flatDeck={flatDeck}
-          />
+      <div role='none' className={mx(deckGrid, 'grid-cols-1 bs-full')}>
```
Comment (thure) [inline packages/plugins/plugin-deck/src/components/DeckLayout/ComplementarySidebar.tsx:41]: `deckGrid` in the sidebar? Is it for the rows?

**Review by thure — APPROVED** [review]
lgtm! just a question about the deck grid.

File: packages/plugins/plugin-deck/src/components/DeckLayout/ComplementarySidebar.tsx (hunk 1 above)
Comment (wittjosiah) [reply packages/plugins/plugin-deck/src/components/DeckLayout/ComplementarySidebar.tsx:41]: question back at you...not sure, this is just what was here 😅

File: packages/plugins/plugin-deck/src/components/DeckLayout/ComplementarySidebar.tsx [hunk 2]
```diff
@@ -18,41 +19,45 @@ import { useNode, useNodeActionExpander } from '../../hooks';
...
   layoutParts: LayoutParts;
   flatDeck?: boolean;
 };
 
-export const ComplementarySidebar = ({ id, layoutParts, flatDeck }: ComplementarySidebarProps) => {
+export const ComplementarySidebar = ({ context, layoutParts, flatDeck }: ComplementarySidebarProps) => {
   const { popoverAnchorId } = useLayout();
+  const attentionPlugin = useResolvePlugin(parseAttentionPlugin);
+  const attention = attentionPlugin?.provides.attention;
+  const attended = attention?.attended ? Array.from(attention.attended) : [];
```
Comment (ZaymonFC) [inline packages/plugins/plugin-deck/src/components/DeckLayout/ComplementarySidebar.tsx:31]: Let's make a hook called `useAttendedIds`. I've seen this repeated a few times now. 
Could put it here: https://github.com/dxos/dxos/blob/0973be1883a944ea2d1caa8e2515e7eaddef8fc0/packages/ui/react-ui-attention/src/components/AttentionProvider.tsx#L33

File: packages/plugins/plugin-deck/src/components/DeckLayout/ComplementarySidebar.tsx (hunk 1 above)
Comment (thure) [reply packages/plugins/plugin-deck/src/components/DeckLayout/ComplementarySidebar.tsx:41]: Oh, derp, right, this was just a change in the ternary expression. Nevermind 😅

**Review by ZaymonFC — APPROVED** [review]
LGTM, will the following code need to change?

https://github.com/dxos/dxos/blob/0973be1883a944ea2d1caa8e2515e7eaddef8fc0/packages/plugins/plugin-thread/src/ThreadPlugin.tsx#L111-L129

**Comment by wittjosiah** [comment]
@ZaymonFC I think that should be unneeded now... I meant to remove it. Will test without it before landing

**Pull request merged**